docker/cli
# Create standalone e2e test image

**- What I did**
- Build a standalone e2e test image that can be run against a cluster
- Add `docker stack` e2e tests for Kubernetes

**- How I did it**

Test modifications:
- Skip tests that require engine configuration modifications when running against a remote endpoint.
- Be less strict about CLI outputs to accommodate running the tests against a Swarm classic endpoint.
- Refactor stack tests so that we run the same test against Swarm and Kubernetes.
- Skip Kubernetes tests unless we have a `KUBECONFIG`.
- Add ability to skip tests that manipulate plugins so that this can be used against EE.

e2e image modifications:

- Do not require mounting volumes to run – image is now completely self-contained.

**- How to verify it**
- The existing flow for running e2e tests should work as before: `make -f docker.Makefile test-e2e`
- Run the image against a cluster running swarm: `docker run --rm -v /path/to/certs:/bundle -e DOCKER_HOST=tcp://host_ip:2376 -e DOCKER_CERT_PATH=/bundle -e DOCKER_TLS_VERIFY=1 -e REMOTE_DAEMON=1 -e SKIP_PLUGIN_TESTS=1 docker-cli-e2e`
- Run the image against a cluster running EE: `docker run --rm -v /path/to/certs:/bundle -e DOCKER_HOST=tcp://host_ip:443 -e DOCKER_CERT_PATH=/bundle -e DOCKER_TLS_VERIFY=1 -e REMOTE_DAEMON=1 -e SKIP_PLUGIN_TESTS=1 docker-cli-e2e`

**- Description for the changelog**

N/A testing only.
